Refresh template using EEA template API

fxprunayre opened the pull request:
Add the possibility to use EEA template API mode and refresh it when required.

In case the API breaks the app (eg. CSS conflicts), catalogue administrator can still access the application with the ```noeeaapi``` URL parameter (http://localhost:8080/geonetwork/srv/eng/catalog.search?noeeaapi) and then turn the API mode off if needed from the admin > settings page.

![image](https://user-images.githubusercontent.com/1701393/43953363-f88f70b0-9c98-11e8-99ff-feb519dbfd2f.png)

Commit:
API loading of the EEA template

Commit:
Changing method signature according to requirements

fxprunayre committed:
EEA API does not return JSON but almost XHTML.

fxprunayre committed:
Use EEA API for baselayout.

fxprunayre commented on services/src/main/java/org/fao/geonet/api/eealayoutapi/EEALayoutManager.java:
```diff
@@ -0,0 +1,228 @@
+
+                sb.append(inline);
+            }
+            sb.append("<script src=\"../../srv/api/eealayoutupdate/script\" /></head>");
```
Why ? @PascalLike

PascalLike replied:
Because the javascript contains characters not compatible with the XSLT engine. In this way, we can insert the javascript code externally from the HTML and avoid to break the XSLT engine

fxprunayre replied:
Ok but we get 
![image](https://user-images.githubusercontent.com/1701393/43948151-50d39882-9c8a-11e8-9540-2805773cb2c4.png)
for this script URL

PascalLike replied:
It's weird because if you access the URL directly the content is returned.

fxprunayre replied:
It's better to put this part directly in the HTML page I think. Doing it

fxprunayre committed:
EEA API / Add parameter to disable it in case it makes the UI unusable. Fallback to current UI.

fxprunayre committed:
Add admin option to turn on/off API mode. In case the API breaks the UI, user can use the noeeaapi URL parameter to access the admin and then turn off the API mode.

fxprunayre committed:
EEA Template API / Add minimal scripts in the HTML page.

MichelGabriel committed:
Fix for broken EEA layout

Changes:
- hide sidepanels
- better color of links in the topbar
- margin top for the GN part of the site

fxprunayre commented:
LGTM, Thanks @MichelGabriel & @PascalLike. Merging.
